2008-06-27  Guillaume Knispel <gknispel@proformatique.com>

	pf_fastagi_args.patch
	$Revision$
	$Date$
	FastAgi interface change (backport from Asterisk 1.6 for 1.4)

	XIVO needs this Asterisk 1.6 style fastagi interface, so we backported
	it for Asterisk 1.4.

	It was first done for Asterisk 1.2 by Richard Braun, I just refreshed
	it for Asterisk 1.4.

Index: asterisk-1.4.21/res/res_agi.c
===================================================================
--- asterisk-1.4.21.orig/res/res_agi.c	2008-06-27 20:08:44.000000000 +0200
+++ asterisk-1.4.21/res/res_agi.c	2008-06-27 20:08:55.000000000 +0200
@@ -143,7 +143,7 @@
 	FastAGI defaults to port 4573 */
 static enum agi_result launch_netscript(char *agiurl, char *argv[], int *fds, int *efd, int *opid)
 {
-	int s;
+	int i, s;
 	int flags;
 	struct pollfd pfds[1];
 	char *host;
@@ -228,6 +228,10 @@
 	if (!ast_strlen_zero(script))
 		fdprintf(s, "agi_network_script: %s\n", script);
 
+	/* argv[1] is either a string or NULL, but always exists */
+	for (i = 1; argv[i] != NULL; i++)
+		fdprintf(s, "agi_arg_%d: %s\n", i, argv[i]);
+
 	if (option_debug > 3)
 		ast_log(LOG_DEBUG, "Wow, connected!\n");
 	fds[0] = s;
